cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -129,11 +129,19 @@ func (cc *createClusterOptions) createCluster(cmd *cobra.Command, _ []string) er
 		return err
 	}
 
-	deps, err := dependencies.ForSpec(ctx, clusterSpec).WithExecutableMountDirs(cc.mountDirs()...).
+	var dirs []string
+	dirs = cc.mountDirs()
+	if clusterSpec.FluxConfig != nil && clusterSpec.FluxConfig.Spec.Git != nil {
+		privateKeyFilePath, fileExists := mountPrivateKeyFile()
+		if fileExists {
+			dirs = append(dirs, privateKeyFilePath)
+		}
+	}
```

Comment:
We are probably going to need this again (upgrade?) so I would vote to make it a bit more reusable

Ideally we would have on single method that, upon receiving all the necessary info, it returns the directories that need to be mounted

we can have something like
`func directoriesToMount(clusterOptions, clusterConfig)`